subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/VerificationFailureHandlingIntegrationTest.groovy

```diff
@@ -0,0 +1,603 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.Matchers
+
+class VerificationFailureHandlingIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
These tests really should be simplified. It is very hard to track what cases they cover. I think there three cases and and all the variations of those should be handled by `where` clauses:
- task action fails with non-verification failure so dependent task does not run
- task action fails with verification failure without `--continue`
- task action fails with verification failure with `--continue

I don't understand why this intermediate task cases add more coverage.